### PR TITLE
[Data objects] Load "Master" as default layout, not "Master (admin mode)"

### LINKED ISSUE #14241: [Bug][Data Objects]: Default layout for admin users
### Expected behavior

Master Layout is the default layout for admin users

### Actual behavior

Master Layout (Admin Mode) is the default layout for admin users

### Steps to reproduce

Related to #14160
https://github.com/pimcore/pimcore/blob/v10.5.16/bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php#L516

This change affects which layout is selected by default for admin users.

Screenshot from 2023-02-06 14-52-32

* index `-1` refers to Master Layout (Admin Mode)
* index `0` refers to Master Layout
* indices `1` onwards are custom layouts

Not sure if this effect was intentional. If it is, can we make it configurable? The reason for defaulting to Master Layout is that the Admin Mode layout makes it unnecessarily easy for users to input invalid data into non-editable fields.

### LINKED ISSUE #14280: [Bug]: Every field is editable as admin
### Expected behavior

Invisible and not editable fields in data object should be marked as them, even if you are logged in as admin.

### Actual behavior

After https://github.com/pimcore/pimcore/pull/14160 the `getSuperLayoutDefinition` is called and everything is editable.

### Steps to reproduce

Use Pimcore 10.5.16:

* Login as admin
* Create a data object with a field which is either not editable or invisible
* Every e.g. input field looks the same, even it shouldn't be edited because it's eg handled by an import.

Field with the settings:
Screenshot from 2023-02-09 15-03-10

Logged in as admin:
Screenshot from 2023-02-09 15-02-55

Logged in as user:
Screenshot from 2023-02-09 15-04-04

### PR DESCRIPTION
**Pull request by BlackbitDevs**
See https://github.com/pimcore/pimcore/pull/14160#issuecomment-1424452554

Resolves #14280 
Resolves #14241

### CODE REVIEW

**Comment by BlackbitDevs** [comment]
@blankse Do you have any idea what PhpStan wants here? I have thought using string keys '0' and '-1' would be enough - but it is not...

**Review by mattamon — COMMENTED** [review]
I had a similar issue yesterday. Stan complained until I removed the condition completely. 

What about renaming the layouts to `'master'` and `'masteradmin'` if we use string anyways. 
Would be more consistent with the custom layouts?

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php [hunk 1]
```diff
@@ -513,7 +513,10 @@ static function (Task $task) {
             }
 
             if ($currentLayoutId === null && count($validLayouts) > 0) {
-                $currentLayoutId = reset($validLayouts)->getId();
+                $currentLayout = $validLayouts['0'] ?? reset($validLayouts);
```
Comment (mattamon) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php]: If we change the ids we then have to check for master here right?
```suggestion
                $currentLayout = $validLayouts['master'] ?? reset($validLayouts);
```

**Comment by blankse** [comment]
Can we not change the order in the `Service::getValidLayouts()`. So `reset` is enough?

**Comment by mattamon** [comment]
@blankse yeah I thought about this too.
Problem is, normally the `master` layout is always available. If you e.g. create two custom views, define none of them as `default` and give a user only these two layouts the `master` layout won't be in the array. Since PhpStan cannot test this I guess, the master check is always true. 
That is what I was trying to fix in the first place.

**Comment by blankse** [comment]
@mattamon It will be fixed if the `Service::getValidLayouts()` adds first the `master` layout, than the `custom layouts` and at last the `masterAdmin` layout, right?

**Comment by mattamon** [comment]
@blankse, I think so yes, this would solve it and we could use reset only. The only thing I need to test, is what it would look like in the list in the backend. Normally the master admin layout is above the master. 

Atm nothing shows up at all, which is kind of a problem I guess :D

image

**Comment by BlackbitDevs** [comment]
Ok, I exchanged order in `getValidLayouts()` and now use `reset()`again. 
For the order of available layouts I have applied a `usort()` function which sorts "Master (admin mode)" first, "Master" second and all other layouts by their name.

**Comment by robertSt7** [comment]
@BlackbitDevs Thanks a lot

**Pull request merged**